proteome-rest/src/main/java/org/uniprot/api/proteome/service/GeneCentricService.java

```diff
@@ -20,40 +17,27 @@
  * @date: 30 Apr 2019
  */
 @Service
-public class GeneCentricService {
-    private final GeneCentricSortClause solrSortClause;
-    private GeneCentricFacetConfig facetConfig;
-    private final BasicSearchService<CanonicalProtein, GeneCentricDocument> basicService;
-    private final DefaultSearchHandler defaultSearchHandler;
+public class GeneCentricService extends BasicSearchService<GeneCentricDocument, CanonicalProtein> {
+    private static final Supplier<DefaultSearchHandler> handlerSupplier =
```

Comment:
I am not sure if this is very helpful to make the code easier to understand. if it is easier to understand for you, I am fine.